File: bank-service/src/main/java/rs/edu/raf/service/CreditTypeService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.CreditTypeCreateDTO;
import rs.edu.raf.dto.CreditTypeDTO;

import java.util.List;

public interface CreditTypeService {
    /**
     * Creates credit type
     * @param creditTypeCreateDTO DTO which contains information about credit type
     * @return {@link CreditTypeDTO} object representing created credit type
     */
    CreditTypeDTO createCreditType(CreditTypeCreateDTO creditTypeCreateDTO);

    /**
     * Gets all credit types
     * @return List of {@link CreditTypeDTO} representing credit types
     */
    List<CreditTypeDTO> getAllCreditTypes();

    /**
     * Deletes credit type
     * @param id the id of the credit type to be deleted
     */
    void deleteCreditType(Long id);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/CreditTypeService.kt
package org.example.bankservicekotlin.service

import org.example.bankservicekotlin.dto.CreditTypeCreateDTO
import org.example.bankservicekotlin.dto.CreditTypeDTO

interface CreditTypeService {
    /**
     * Creates credit type
     * @param creditTypeCreateDTO DTO which contains information about credit type
     * @return [CreditTypeDTO] object representing created credit type
     */
    fun createCreditType(creditTypeCreateDTO: CreditTypeCreateDTO): CreditTypeDTO

    /**
     * Gets all credit types
     * @return List of [CreditTypeDTO] representing credit types
     */
    fun getAllCreditTypes(): List<CreditTypeDTO>

    /**
     * Deletes credit type
     * @param id the id of the credit type to be deleted
     */
    fun deleteCreditType(id: Long)
}